### PR TITLE
Introduce Docker debugging with Docker Compose orchestration support

### PR DESCRIPTION
**Pull request by cheenamalhotra**
This PR introduces:
1. A new **Microsoft.Data.SqlClient.DockerLinuxTest** Project that contains basic app that enables debugging SqlClient (NetCore) in Linux containers from Windows. Docker support also added with .NET Core SDK - 3.1-buster docker image.
2. Docker-compose support with the above **DockerLinuxTest** project, by connecting to SQL Server 2019 hosted from another docker container.

Note to reviewers:
* No code changes are needed to run this test.
* Run "docker-compose" as startup project and configuration, expect below output in Debug Window:
`Connected to SQL Server v15.00.4023 from Unix 4.19.76.0`
* Breakpoints can be set in the test project or SqlClient (NetCore) driver as needed to debug Unix.

Also includes 'doc' folder and files in solution for handy editing experience.

/cc: @Wraith2

### CODE REVIEW

File: src/Microsoft.Data.SqlClient/tests/DockerLinuxTest/Program.cs [hunk 1]
```diff
@@ -0,0 +1,28 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using Microsoft.Data.SqlClient;
+
+namespace Microsoft.Data.SqlClient.DockerLnuxTest
```
Comment (DavoudEshtehari) [inline src/Microsoft.Data.SqlClient/tests/DockerLinuxTest/Program.cs]: Just DockerL**i**nuxTest.

File: src/Microsoft.Data.SqlClient/netcore/src/Microsoft.Data.SqlClient.csproj [hunk 2]
```diff
@@ -606,7 +606,7 @@
       <CustomToolNamespace>System</CustomToolNamespace>
     </EmbeddedResource>
   </ItemGroup>
-  <Import Project="$(ToolsDir)targets\GenerateThisAssemblyCs.targets" />
-  <Import Project="$(ToolsDir)targets\ResolveContract.targets" Condition="'$(OSGroup)' == 'AnyOS'" />
-  <Import Project="$(ToolsDir)targets\NotSupported.targets" Condition="'$(OSGroup)' == 'AnyOS'" />
+  <Import Project="tools\targets\GenerateThisAssemblyCs.targets" />
+  <Import Project="tools\targets\ResolveContract.targets" Condition="'$(OSGroup)' == 'AnyOS'" />
+  <Import Project="tools\targets\NotSupported.targets" Condition="'$(OSGroup)' == 'AnyOS'" />
```
Comment (David-Engel) [inline src/Microsoft.Data.SqlClient/netcore/src/Microsoft.Data.SqlClient.csproj]: Are these changes on purpose?

Comment (cheenamalhotra) [reply src/Microsoft.Data.SqlClient/netcore/src/Microsoft.Data.SqlClient.csproj]: Yes, I did it to keep tools alongwith netcore project, not mandatory since I managed to get the whole folder copied later.. we can choose eitherway.

File: tools/credScan/CredScanSuppressions.json [hunk 3]
```diff
@@ -20,6 +20,14 @@
...
+	{
+      "file": "Program.cs",
+      "_justification": "Contains dummy passwords used for docker testing"
+    },
+	{
+      "file": "docker-compose.yml",
+      "_justification": "Contains dummy passwords used for docker testing"
     }
   ]
 }
```
Comment (JRahnama) [inline tools/credScan/CredScanSuppressions.json]: Please add EOF line here.

File: src/Microsoft.Data.SqlClient/netcore/src/Microsoft.Data.SqlClient.csproj (hunk 2 above)
Comment (cheenamalhotra) [reply src/Microsoft.Data.SqlClient/netcore/src/Microsoft.Data.SqlClient.csproj]: Moved back to their original location.

File: src/docker-compose.dcproj [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+    <DockerServiceName>microsoft.data.sqlclient.dockertests</DockerServiceName>
+  </PropertyGroup>
+  <ItemGroup>
+    <None Include="docker-compose.override.yml">
+      <DependentUpon>docker-compose.yml</DependentUpon>
+    </None>
+    <None Include="docker-compose.yml" />
+    <None Include=".dockerignore" />
+  </ItemGroup>
+</Project>
```
Comment (David-Engel) [inline src/docker-compose.dcproj]: Will adding this project type change any requirements in our BUILDGUIDE? (I don't have Docker installed, for example.)

Comment (cheenamalhotra) [reply src/docker-compose.dcproj]: Building & Testing should be fine but for seamless experience in Visual Studio I can update the vsconfig files to include docker too.

Comment (ErikEJ) [reply src/docker-compose.dcproj]: Yes, do not expect everyone to have Docker Tools installed

Comment (cheenamalhotra) [reply src/docker-compose.dcproj]: I verified and our VS19Components.vsconfig includes "Container Tools" as tested on v16.5.1.

File: src/docker-compose.dcproj [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+    <DockerServiceName>microsoft.data.sqlclient.dockertests</DockerServiceName>
+  </PropertyGroup>
+  <ItemGroup>
+    <None Include=".dockerignore" />
+    <None Include="docker-compose.override.yml">
+      <DependentUpon>docker-compose.yml</DependentUpon>
+    </None>
+    <None Include="docker-compose.yml" />
+  </ItemGroup>
+</Project>
```
Comment (DavoudEshtehari) [inline src/docker-compose.dcproj:18]: The EOF is not consider.
